scipy/stats/_continuous_distns.py

```diff
@@ -7943,27 +7944,112 @@ class argus_gen(rv_continuous):
+        A = chi**3 / (_norm_pdf_C * _argus_phi(chi))
+        return A * x * np.sqrt(y) * np.exp(-chi**2 * y / 2)
 
     def _cdf(self, x, chi):
-        """
-        Return CDF of the argus function
-        """
         return 1.0 - self._sf(x, chi)
 
     def _sf(self, x, chi):
-        """
-        Return survival function of the argus function
-        """
         return _argus_phi(chi * np.sqrt(1 - x**2)) / _argus_phi(chi)
 
+    def _rvs(self, chi):
+        chi = np.asarray(chi)
+        if chi.size == 1:
+            out = self._rvs_scalar(chi, self._size)
+        else:
+            shp, bc = _check_shape(chi.shape, self._size)
+            numsamples = int(np.prod(shp))
+            out = np.empty(self._size)
+            it = np.nditer([chi],
+                           flags=['multi_index'],
+                           op_flags=[['readonly']])
+            while not it.finished:
+                idx = tuple((it.multi_index[j] if not bc[j] else slice(None))
+                            for j in range(-len(self._size), 0))
+                r = self._rvs_scalar(it[0], numsamples)
+                out[idx] = r.reshape(shp)
+                it.iternext()
+
+        if self._size == ():
+            out = out[()]
+        return out
+
+    def _rvs_scalar(self, chi, numsamples=None):
+        # if chi <= 2.611:
+        # use rejection method, see Devroye:
+        # Non-Uniform Random Variate Generation, 1986, section II.3.2.
+        # write: self.pdf = c * g(x) * h(x), where
+        # h is [0,1]-valued and g is a density
+        # g(x) = d1 * chi**2 * x * exp(-chi**2 * (1 - x**2) / 2), 0 <= x <= 1
+        # h(x) = sqrt(1 - x**2), 0 <= x <= 1
+        # Integrating g, we get:
+        # G(x) = d1 * exp(-chi**2 * (1 - x**2) / 2) - d2
+        # d1 and d2 are determined by G(0) = 0 and G(1) = 1
+        # d1 = 1 / (1 - exp(-0.5 * chi**2))
+        # d2 = 1 / (exp(0.5 * chi**2) - 1)
+        # => G(x) = (exp(chi**2 * x**2 /2) - 1) / (exp(chi**2 / 2) - 1)
+        # expected number of iterations is c with
+        # c = -np.expm1(-0.5 * chi**2) * chi / (_norm_pdf_C * _argus_phi(chi))
+        # note that G can be inverted easily, so we can sample
+        # rvs from this distribution
+        # G_inv(y) = sqrt(2 * log(1 + (exp(chi**2 / 2) - 1) * y) / chi**2)
+        # to avoid an overflow of exp(chi**2 / 2), it is convenient to write
+        # G_inv(y) = sqrt(1 + 2 * log(exp(-chi**2 / 2) * (1-y) + y) / chi**2)
+        #
+        # if chi > 2.611:
+        # use ratio of uniforms method applied to a transformed variable of X
+        # (X is ARGUS with parameter chi):
+        # Y = chi * sqrt(1 - X**2) has density proportional to
+        # u**2 * exp(-u**2 / 2) on [0, chi] (Maxwell distribution conditioned
+        # on [0, chi]). Apply ratio of uniforms to this density to generate
+        # samples of Y and convert back to X
+        # the expected number of iterations is below 2.2 for all chi > 0
+
+        if chi <= 2.611:
```

Comment:
maybe add a line w/comment describing the fixed point equation that gives the `chi` value?

Reply:
added

Reply:
Maybe something like: 
```
# the value 2.611 can be found by comparing the rejection constants
# of the ratio of uniforms method and the rejection method which gives
# chi*(1 - exp(- 0.5*chi**2)) = 8*sqrt(2)*exp(-1.5), solving for chi gives ~2.61
```
That gives the actual formula for quick reference

Reply:
added